odl/solvers/functional/functional.py

```diff
@@ -0,0 +1,1671 @@
+        -----
+        The convex conjugate functional of a convex functional :math:`f(x)`,
+        defined on a Hilber space, is defined as the functional
+
+        .. math::
+
+            f^*(x^*) = \\sup_{x} \{ \\langle x^*,x \\rangle - f(x)  \}.
+
+        The concept is also known as the Legendre transformation.
+
+        References
+        ----------
+        Wikipedia article on `Convex conjugate
+        <https://en.wikipedia.org/wiki/Convex_conjugate>`_.
+
+        Wikipedia article on `Legendre transformation
+        https://en.wikipedia.org/wiki/Legendre_transformation
+
+        For literature references see, e.g., [Lue1969]_, [Roc1970]_.
+        """
+        return self.operator.conjugate_functional * (1.0 / self.vector)
+
+
+class FunctionalSum(Functional, OperatorSum):
+
+    """Expression type for the sum of functionals.
+
+    ``FunctionalSum(func1, func2) <==> (x --> func1(x) + func2(x))``.
+    """
+
+    def __init__(self, func1, func2, tmp_dom=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        func1, func2 : `Functional`
+            The summands of the functional sum. Their `Operator.domain`
+            and `Operator.range` must coincide.
+        tmp_dom : `Operator.domain` `element`, optional
+            Used to avoid the creation of a temporary when applying the
+            gradient.
+        """
+        # TODO: Tmp_dom, what is it used for?
+        if not isinstance(func1, Functional):
+            raise TypeError('functional 1 {!r} is not a Functional instance.'
+                            ''.format(func1))
+        if not isinstance(func2, Functional):
+            raise TypeError('functional 2 {!r} is not a Functional instance.'
+                            ''.format(func2))
+
+        Functional.__init__(self, domain=func1.domain,
+                            linear=(func1.is_linear and func2.is_linear),
+                            grad_lipschitz=(func1.grad_lipschitz +
+                                            func2.grad_lipschitz))
+
+        OperatorSum.__init__(self, func1, func2, tmp_ran=None,
+                             tmp_dom=tmp_dom)
+
+    @property
+    def gradient(self):
+        """Gradient operator of functional sum."""
+        return self.left.gradient + self.right.gradient
+
+    def derivative(self, point):
+        """Returns the derivative operator in the given point.
+
+        This function returns the linear operator
+
+            ``x --> <x, grad_f(point)>``,
+
+        where ``grad_f(point)`` is the gradient of the functional in the point
+        ``point``.
+
+        Parameters
+        ----------
+        point : `LinearSpaceVector`
+            The point in which the gradient is evaluated.
+
+        Returns
+        -------
+        out : `DerivativeOperator`
+            The linear operator that maps ``x --> <x, grad_f(point)>``.
+        """
+        return self._op1.derivative(point) + self._op2.derivative(point)
```

Comment:
So remove the method completely? It will fall back on `OperatorSum`, so I guess it gets the same doc as in `OperatorSum`:

```
    Return the operator derivative at ``x``.

    The derivative of a sum of two operators is equal to the sum of
    the derivatives.
    
    Parameters
    ----------
    x : `domain` `element-like`
        Evaluation point of the derivative
```

Is that ok?